src/main/java/com/google/devtools/build/lib/remote/RemoteModule.java

```diff
@@ -929,7 +929,14 @@ public void executorInit(CommandEnvironment env, BuildRequest request, ExecutorB
       remoteOutputService.setActionInputFetcher(actionInputFetcher);
       actionContextProvider.setActionInputFetcher(actionInputFetcher);
 
-      if (remoteOutputsMode.downloadToplevelOutputsOnly()) {
+      boolean useToplevelDownloader = remoteOutputsMode.downloadToplevelOutputsOnly();
+
+      // Download toplevel artifacts for `run` command.
+      if ("run".equals(env.getCommandName())) {
```

Comment:
We might need to support other commands (e.g. `coverage`, `mobile-install`, etc.). Do you think whether it is better if we **always** use toplevel downloader but pass in the download mode, let the downloaders decide what to download in both minimal/toplevel mode?

Reply:
Done.